### PR TITLE
Some THM cleanup

### PR DESCRIPTION
**Pull request by joshuahansel**
This PR:

- Fixes/adds some THM system documentation
- Removes model-specific (like 2-phase) check methods from `Component`
- Moves model-specific parameters from `THMProblem` to the respective components

### CODE REVIEW

**Comment by joshuahansel** [comment]
The RELAP-7 PR is a pre-requisite: https://hpcgitlab.hpc.inl.gov/idaholab/relap-7/-/merge_requests/1857. Other apps may need changes (we'll see).

File: modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/syntax/ControlLogic/index.md [hunk 1]
```diff
@@ -1 +1,25 @@
-!template load file=stubs/moose_system.md.template name=ControlLogic syntax=/ControlLogic
+# ControlLogic System
+
+The `ControlLogic` system is an extension of MOOSE's [Controls system](syntax/Controls/index.md).
+Standard MOOSE `Control`s can be created with the `ControlLogic` syntax without
+effect. `THMControl`s, however, which have additional functionality, require
```
Comment (licharlot) [inline modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/syntax/ControlLogic/index.md:5]: Delete "without effect".

**Review by licharlot — CHANGES_REQUESTED** [review]

**Review by licharlot — DISMISSED** [review]

File: modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/syntax/ControlLogic/index.md [hunk 2]
```diff
@@ -1 +1,25 @@
-!template load file=stubs/moose_system.md.template name=ControlLogic syntax=/ControlLogic
+# ControlLogic System
+
+The `ControlLogic` system is an extension of MOOSE's [Controls system](syntax/Controls/index.md).
+Standard MOOSE `Control`s can be created with the `ControlLogic` syntax.
+`THMControl`s, however, which have additional functionality, require
+the `ControlLogic` syntax.
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/syntax/ControlLogic/index.md:6]: this read weird. 
Do you really mean it s just an option in MOOSE and mandatory in THM

Comment (joshuahansel) [reply modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/syntax/ControlLogic/index.md:6]: I'm trying to say that you can do this
```
[ControlLogic]
  [some_moose_control]
    ...
  []
  [some_thm_control]
    ...
  []
[]
```
but not this:
```
[Controls]
  [some_moose_control]
    ...
  []
  [some_thm_control] # this is an error
    ...
  []
[]
```

**Comment by joshuahansel** [comment]
I'll wait to merge this until the multi-apps get RELAP-7 and Sockeye updates so that we can avoid breaking them.

**Review by licharlot — APPROVED** [review]

**Pull request merged**